### PR TITLE
fix(git/auth): Fix submodule authentication with gitlab

### PR DESCRIPTION
**Pull request by uncycler**
## Changes

Add https://oauth2@<hostname> in git authenticationRules because gitlab platform hardcode this value. Without it, the insteadOf is not matched

## Context

Since this PR (https://github.com/renovatebot/renovate/pull/13477), gitlab submodule scanning is broken and git ask a password for the oauth2 user.

## Documentation

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [X] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by Shegox** [comment]
Probably addresses #24077 as well

**Review by viceice — COMMENTED** [review]
failed tests and lint issues

File: lib/util/git/auth.ts [hunk 1]
```diff
@@ -138,6 +138,16 @@ export function getAuthenticationRules(
     }/${insteadUrl.full_name}${insteadUrl.git_suffix ? '.git' : ''}`,
   });
 
+  // When on gitlab, we add oauth2 since it's hardcoded.
+  if (token.startsWith('gitlab-ci-token')) {
+    authenticationRules.push({
+      url: url.toString(protocol),
+      insteadOf: `https://oauth2@${insteadUrl.resource}${
+        insteadUrl.port ? `:${insteadUrl.port}` : ''
+      }/${insteadUrl.full_name}${insteadUrl.git_suffix ? '.git' : ''}`,
```
Comment (viceice) [inline lib/util/git/auth.ts:147]: you need more test cases for this to cover all branches (conditions).

you can probably compute this part once and reuse it. then you don't need a new test. 😉

**Comment by Shegox** [comment]
I think #24306 might address the same problem.

**Comment by rarkins** [comment]
@uncycler are you able to add the necessary tests?

**Comment by rarkins** [comment]
Closing due to inactivity - PR is welcome to be resumed later

**Pull request closed without merging**